### PR TITLE
Refactor IAP

### PR DESCRIPTION
**Pull request by blackzlq**
Refactor IAP module

Tested-by: zlq on AP cluster

### CODE REVIEW

File: modules/iap/iap.tf [hunk 1]
```diff
@@ -19,145 +19,72 @@ resource "random_string" "random" {
...
+  name         = "${var.app_name}-address-${random_string.random.result}"
   address_type = "EXTERNAL"
   ip_version   = "IPV4"
 }
 
 # Helm Chart IAP
 resource "helm_release" "iap_frontend" {
-  count            = var.frontend_add_auth ? 1 : 0
-  name             = "iap-frontend"
+  count            = var.add_auth ? 1 : 0
```
Comment (umeshkumhar) [inline modules/iap/iap.tf:34]: add_auth is not required here.. module will always be called when add_auth is true

File: modules/jupyter/main.tf [hunk 2]
```diff
@@ -46,19 +46,20 @@ module "iap_auth" {
...
-  jupyter_k8s_backend_service_name = var.k8s_backend_service_name
-  jupyter_k8s_backend_service_port = var.k8s_backend_service_port
-  jupyter_client_id                = var.client_id != "" ? var.client_id : google_iap_client.iap_oauth_client[0].client_id
-  jupyter_client_secret            = var.client_id != "" ? var.client_secret : google_iap_client.iap_oauth_client[0].secret
-  jupyter_url_domain_addr          = var.url_domain_addr
-  jupyter_url_domain_name          = var.url_domain_name
+  project_id               = var.project_id
+  namespace                = var.namespace
+  app_name                 = "jupyter"
+  add_auth                 = var.add_auth
```
Comment (umeshkumhar) [inline modules/jupyter/main.tf:52]: this inputs is not required. as count is already set

File: applications/rag/frontend/main.tf [hunk 3]
```diff
@@ -42,19 +42,20 @@ module "iap_auth" {
...
+  namespace                = var.namespace
+  app_name                 = "frontend"
+  add_auth                 = var.add_auth
+  k8s_ingress_name         = var.k8s_ingress_name
+  k8s_managed_cert_name    = var.k8s_managed_cert_name
+  k8s_iap_secret_name      = var.k8s_iap_secret_name
+  k8s_backend_config_name  = var.k8s_backend_config_name
+  k8s_backend_service_name = var.k8s_backend_service_name
+  k8s_backend_service_port = var.k8s_backend_service_port
+  client_id                = var.client_id != "" ? var.client_id : google_iap_client.iap_oauth_client[0].client_id
```
Comment (umeshkumhar) [inline applications/rag/frontend/main.tf:55]: can we move `resource "google_project_service" "project_service"` and `resource "google_iap_client" "iap_oauth_client"` into iap module? 


just take input of client_id and in module check if its empty then attempt new oauth client creation.

**Comment by imreddy13** [comment]
@blackzlq Please test on both AP and standard clusters. Please don't merge untested changes.

**Pull request closed without merging**